simpleclient/src/main/java/io/prometheus/client/Collector.java

```diff
@@ -140,9 +140,9 @@ public String toString() {
    */
   public static final double MILLISECONDS_PER_SECOND = 1E3;
 
-  protected static final Pattern METRIC_NAME_RE = Pattern.compile("[a-zA-Z_:][a-zA-Z0-9_:]*");
-  protected static final Pattern METRIC_LABEL_NAME_RE = Pattern.compile("[a-zA-Z_][a-zA-Z0-9_]*");
-  protected static final Pattern RESERVED_METRIC_LABEL_NAME_RE = Pattern.compile("__.*");
+  private static final Pattern METRIC_NAME_RE = Pattern.compile("[a-zA-Z_:][a-zA-Z0-9_:]*");
```

Comment:
You have protected-level methods to perform those checks, you don't need access to the regexes.